aproxy.go

```diff
@@ -0,0 +1,404 @@
+//go:build linux
+
+package main
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"flag"
+	"fmt"
+	"golang.org/x/crypto/cryptobyte"
+	"io"
+	"log"
+	"log/slog"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"unsafe"
+)
+
+var version = "0.2.0"
+
+type PrereadConn struct {
+	prereadStarted bool
+	prereadEnded   bool
+	prereadBuf     []byte
+	mu             sync.Mutex
+	conn           net.Conn
+}
+
+func (c *PrereadConn) StartPreread() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.prereadStarted {
+		panic("call StartPreread after preread has already started or ended")
+	}
+	c.prereadStarted = true
+}
+
+func (c *PrereadConn) RestorePreread() {
```

Comment:
It's strange to me that this sets "prereadEnded = true" but is called "restore". Why not `EndPreread` to mirror `StartPreread` and match the field name?

Reply:
Yes, indeed, I refactored this a little bit to remove the `StartPreread` as the `StartPreread` is a little bit redundant and renamed the `RestorePreread` to `EndPreread`.